test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeControllerSpec.scala

```diff
@@ -945,6 +949,197 @@ class RepresenteeControllerSpec
+
+        "the user has no id " when {
+          val requiredPreviousAnswers =
+            IncompleteRepresenteeAnswers(Some(IndividualName("First", "Last")), None, None, None, false)
+          "redirect to enter id page" in {
+            inSequence {
+              mockAuthWithNoRetrievals()
+              mockGetSession(
+                sessionWithStartingNewDraftReturn(
+                  requiredPreviousAnswers,
+                  Left(PersonalRepresentative)
+                )._1
+              )
+            }
+            checkIsRedirect(performAction(), routes.RepresenteeController.checkYourAnswers())
+          }
+        }
+      }
+
+    }
+
+    "handling requests to submit the confirm person page" must {
+      def performAction(formData: Seq[(String, String)]): Future[Result] =
+        controller.confirmPersonSubmit()(FakeRequest().withFormUrlEncodedBody(formData: _*))
+
+      behave like redirectToStartBehaviour(() => performAction(Seq.empty))
+
+      behave like nonCapacitorOrPersonalRepBehaviour(() => performAction(Seq.empty))
+
+      "show a form error" when {
+
+        val session = sessionWithFillingOutReturn(
+          IncompleteRepresenteeAnswers(
+            Some(sample[IndividualName]),
+            Some(sample[RepresenteeNino]),
+            Some(sample[DateOfDeath]),
+            None,
+            false
+          ),
+          Right(Capacitor)
+        )._1
+
+        def testConfirmPersonFormError(
+          performAction: Seq[(String, String)] => Future[Result],
+          currentSession: SessionData,
+          expectedErrorMessageKey: String
+        )(data: Seq[(String, String)]): Unit = {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(currentSession)
+          }
+
+          checkPageIsDisplayed(
+            performAction(data),
+            messageFromMessageKey("representeeConfirmPerson.title"), { doc =>
+              doc.select("#confirmed-inline-error").text() contains messageFromMessageKey(
+                expectedErrorMessageKey
+              )
+              doc.select("#confirmed-inline-error > span").text() shouldBe ("Error:")
+            },
+            BAD_REQUEST
+          )
+        }
+
+        "nothing is selected" in {
+          testConfirmPersonFormError(performAction, session, "confirmed.error.required")(Seq.empty)
+        }
+      }
+
+      "redirect the page to enter-name" when {
+        "the user has clicked No " in {
+          val requiredPreviousAnswers = IncompleteRepresenteeAnswers(
+            Some(IndividualName("First", "Last")),
+            Some(RepresenteeNino(NINO("AB123456C"))),
+            None,
+            None,
+            false
+          )
+
+          val session = sessionWithStartingNewDraftReturn(
+            requiredPreviousAnswers,
+            Left(PersonalRepresentative)
+          )._1
+          val Some(sndr: StartingNewDraftReturn) = session.journeyStatus
```

Comment:
you don't need to extract out the journey status this way - the journey status is returned as part of `sessionWithStartingNewDraftReturn`

Reply:
Done.